src/Engine/Graphics/BspRenderer.cpp

```diff
@@ -86,31 +90,22 @@ void BspRenderer::AddFaceToRenderList_d3d(int node_id, int uFaceID) {
 
         // calculates the portal bounding and frustum
         bool bFrustumbuilt = CalcPortalShapePoly(
-                pFace, static_subAddFaceToRenderList_d3d_stru_F79E08,
+                pFace, clippedFaceVertices,
                 &pNewNumVertices, nodes[num_nodes].ViewportNodeFrustum.data(),
                 nodes[num_nodes].pPortalBounding.data());
 
-        auto boundingMatches = [](const std::array<RenderVertexSoft, 4> &l, const std::array<RenderVertexSoft, 4> &r) {
-            for (int i = 0; i < l.size(); i++) {
-                if (l[i].vWorldPosition != r[i].vWorldPosition)
-                    return false;
-            }
-            return true;
-        };
-
         // avoid circular loops in portals
         for (int test = 0; test < num_nodes; test++) {
-            if (nodes[test].uSectorID == nodes[num_nodes].uSectorID &&
-                nodes[test].uFaceID == nodes[num_nodes].uFaceID &&
-                boundingMatches(nodes[test].pPortalBounding, nodes[num_nodes].pPortalBounding)) {
+            if (nodes[test].uSectorID == nodes[num_nodes].uSectorID) {
                 return;
             }
         }
```

Comment:
We cannot drop this segment. The same sector may be visible through multiple portals so we must make sure if the nodes is unique.

Reply:
OK, based on the save file, this means this part needs further investigation as we shouldn't have the same sector in two nodes (as we would be adding the same BSPnodes and faces there once again), yet we seem to require it there.

I will revert this part and document the new finding.